pkg/services/resource.go

```diff
@@ -131,6 +134,19 @@ func (s *sqlResourceService) Update(ctx context.Context, resource *api.Resource)
 
 	// Increase the current resource version and update its manifest.
 	found.Version = found.Version + 1
+	// Also increase metadata generation to ensure version consistency
+	if metadata, ok := resource.Payload[cetypes.ExtensionWorkMeta]; ok {
+		metaJson, err := json.Marshal(metadata)
```

Comment:
we may use to avoid `Marshal/Unmarshal `
```
metaObj, ok := metadata.(map[string]interface{})
	if !ok {
		return fmt.Errorf("unsupported object type %T", metadata)
	}
	// using
	metaObj["generation"] = int64(found.Version)
```